site/app/libraries/NotebookUtils.php

```diff
@@ -64,21 +87,34 @@ public static function jupyterToSubmittyNotebook(string $filepath): array {
                                 }
                             }
 
+                            $text = $output['data']['text/plain'] ?? '';
+                            $output_text = is_array($text) ? implode($text) : (string) $text;
+
                             if ($output_type === 'text/plain') {
                                 // Display output text if we don't know how to render the content otherwise
                                 $cells[] = [
                                     'type' => 'output',
-                                    'output_text' => $output['data']['text/plain'],
+                                    'output_text' => $output_text,
                                 ];
                             }
                             elseif ($output_type !== null) {
-                                $cells[] = [
-                                    'type' => 'image',
-                                    'image' => 'data:' . $output_type . ';base64, ' . $output['data'][$output_type],
-                                    'width' => 0,
-                                    'height' => 0,
-                                    'alt_text' => $output['data']['text/plain'] ?? '',
-                                ];
+                                $img = $output['data'][$output_type];
+                                if (strlen($img) <= $img_size_limit) {
+                                    $cells[] = [
+                                        'type' => 'image',
+                                        'image' => "data:" . $output_type . ';base64, ' . $img,
+                                        'width' => 0,
+                                        'height' => 0,
+                                        'alt_text' => $output_text,
+                                    ];
+                                }
+                                else {
+                                    $log_message = 'Image skipped: exceeds size limit of ' . $img_size_limit . ' bytes.';
```

Comment:
How hard would it be to give users the ability to click a button to load the image anyway?